rclpy/src/rclpy/_rclpy.c

```diff
@@ -2459,7 +2468,47 @@ rclpy_take(PyObject * Py_UNUSED(self), PyObject * args)
     return PyErr_NoMemory();
   }
 
-  rcl_ret_t ret = rcl_take(subscription, taken_msg, NULL);
+  rcl_ret_t ret;
+  if (raw) {
+    destroy_ros_message(taken_msg);
+    Py_DECREF(pymetaclass);
+
+    // Create a serialized message object
+    // Not 100% sure that 0u is the best size, but it will do
```

Comment:
it will do as the message then gets dynamically resized down the stack when the actual message is taken from the middleware. 
I am not sure where to put this best in the rclpy API, but I guess it would be nice to have some sort of memory allocation system where this can eventually become a parameter to configure.

Reply:
For now I think it's ok that it just uses the default allocator. If we need a more specialized one, it can be added but this functionality will probably always be fairly niche (debugging and diagnostics most likely).